gridstatus/ieso.py

```diff
@@ -0,0 +1,499 @@
+
+        Returns:
+            pd.DataFrame: Ontario load forecast
+        """
+        if isinstance(date, tuple):
+            date, _end = date
+
+        today = utils._handle_date("today", tz=self.default_timezone)
+
+        if date != "latest":
+            date = utils._handle_date(date, tz=self.default_timezone)
+
+            if date.date() < today.date() - pd.Timedelta(
+                days=MAXIMUM_DAYS_IN_PAST_FOR_LOAD_FORECAST,
+            ):
+                raise NotSupported(
+                    "Past dates are not supported for load forecasts more than "
+                    f"{MAXIMUM_DAYS_IN_PAST_FOR_LOAD_FORECAST} days in the past.",
+                )
+
+            if date.date() > today.date() + pd.Timedelta(
+                days=MAXIMUM_DAYS_IN_FUTURE_FOR_LOAD_FORECAST,
+            ):
+                raise NotSupported(
+                    "Load forecasts are not available more than "
+                    f"{MAXIMUM_DAYS_IN_FUTURE_FOR_LOAD_FORECAST} days in the future",
+                )
+
+        root = ET.fromstring(self._request(LOAD_FORECAST_URL, verbose).text)
+
+        # Extract values from <DataSet Series="Projected">
+        projected_values = []
+
+        # Iterate through the XML to find the DataSet with Series="Projected"
+        for dataset in root.iter("DataSet"):
+            if dataset.attrib.get("Series") == "Projected":
+                for data in dataset.iter("Data"):
+                    for value in data.iter("Value"):
+                        projected_values.append(value.text)
+
+        created_at = pd.Timestamp(
+            root.find(".//CreatedAt").text,
+            tz=self.default_timezone,
+        )
+        start_date = pd.Timestamp(
+            root.find(".//StartDate").text,
+            tz=self.default_timezone,
+        )
+
+        # Create the range of interval starts based on the number of values at an
+        # hourly frequency
+        interval_starts = pd.date_range(
+            start_date,
+            periods=len(projected_values),
+            freq="H",
+            tz=self.default_timezone,
+        )
+
+        # Create a DataFrame with the projected values
+        df_projected = pd.DataFrame(projected_values, columns=["Ontario Load Forecast"])
+        df_projected["Ontario Load Forecast"] = df_projected[
+            "Ontario Load Forecast"
+        ].astype(float)
+        df_projected["Publish Time"] = created_at
+        df_projected["Interval Start"] = interval_starts
+        df_projected["Interval End"] = df_projected["Interval Start"] + pd.Timedelta(
+            hours=HOUR_INTERVAL,
+        )
+
+        df = utils.move_cols_to_front(
+            df_projected,
+            [
+                "Interval Start",
+                "Interval End",
+                "Publish Time",
+                "Ontario Load Forecast",
+            ],
+        )
+
+        # Latest returns both today and tomorrow
+        if date == "latest":
```

Comment:
no need to do this filtering here. just return the complete forecast as-is. the caller of the function can easily do this type of filtering if they care.